### PR TITLE
Remove charts/kueue/templates/visibility/kustomization.yaml

### LINKED ISSUE #1459: Helm install Error
**What happened**:
```
[root@test-1 charts]# helm install kueue kueue/ --create-namespace --namespace kueue-system
Error: INSTALLATION FAILED: unable to build kubernetes objects from release manifest:
 error validating "": error validating data: [apiVersion not set, kind not set]
```
**What you expected to happen**:
install success
**How to reproduce it (as minimally and precisely as possible)**:
```
git clone https://github.com/kubernetes-sigs/kueue.git
cd kueue/charts
helm install kueue kueue/ --create-namespace --namespace kueue-system
```
**Anything else we need to know?**:
I tried `helm template ./charts/kueue --values ./charts/kueue/values.yaml` to debug where did this error happen,
then I find out:
```
---
# Source: kueue/templates/visibility/kustomization.yaml
resources:
- apiservice.yaml
- role_binding.yaml
- service.yaml
---
```
This is where helm got Error, so I tried to remove `kueue/templates/visibility/kustomization.yaml`, it works.
So I wonder, should we remove this file?

**Environment**:
- Kubernetes version (use `kubectl version`): v1.26.0
- Kueue version (use `git describe --tags --dirty --always`): main
- Cloud provider or hardware configuration:
- OS (e.g: `cat /etc/os-release`):
- Kernel (e.g. `uname -a`):
- Install tools: Helm
- Others: helm version: v3.9.0

### PR DESCRIPTION
**Pull request by B1F030**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
remove charts/kueue/templates/visibility/kustomization.yaml, and modify the [hack/update-helm.sh](https://github.com/kubernetes-sigs/kueue/blob/main/hack/update-helm.sh).
#### Which issue(s) this PR fixes:

Fixes #1459

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

File: hack/update-helm.sh [hunk 1]
```diff
@@ -33,7 +33,7 @@ EXCLUDE_FILES='kustomization.yaml'
 
 # Copy all YAML files from the source directory to the destination directory
 cp ${SRC_CRD_DIR}/*.yaml ${DEST_CRD_DIR}
-cp ${SRC_VISIBILITY_DIR}/*.yaml ${DEST_VISIBILITY_DIR}
+find $SRC_VISIBILITY_DIR -name "*.yaml" $(printf "! -name %s " $EXCLUDE_FILES) -exec cp "{}" $DEST_VISIBILITY_DIR \;
```
Comment (tenzen-y) [inline hack/update-helm.sh:36]:
```suggestion
find ${SRC_VISIBILITY_DIR} -name "*.yaml" $(printf "! -name %s " ${EXCLUDE_FILES}) -exec cp "{}" ${DEST_VISIBILITY_DIR} \;
```

nit

**Review by tenzen-y — COMMENTED** [review]
otherwise lgtm

File: hack/update-helm.sh (hunk 1 above)
Comment (B1F030) [reply hack/update-helm.sh:36]: [L37](https://github.com/kubernetes-sigs/kueue/blob/main/hack/update-helm.sh#L37)
There was no '{}', so I think it's ok not to add '{}', better keep it same, WDYT?

Comment (tenzen-y) [reply hack/update-helm.sh:36]: Oh, that makes sense. Thanks.

**Review by tenzen-y — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**